types/main.d.ts

```diff
@@ -0,0 +1,624 @@
+         * default: pdfjs v1.3 (github.com/rkusa/pdfjs)
+         */
+        producer?: string;
+        /**
+         * The date and time the document was created
+         * default: new Date()
+         */
+        creationDate?: Date;
+        /**
+         * The date and time the document was most recently modified
+         */
+        modDate?: Date;
+    };
+
+    export class Document extends Fragment {
+        /**
+         * Creates a new PDF document
+         * @param opts document options
+         */
+        constructor(opts: DocumentOptions);
+
+        /**
+         * Get document's properties
+         */
+        info: DocumentProperties & { id: string };
+
+        /**
+         * Document is a Readable stream and can therefore piped into other streams
+         * @param dest destination
+         * @param opts options
+         */
+        pipe<T>(dest: T, opts?: { end?: boolean }): T;
+
+        on(event: string | symbol, listener: (...args: any[]) => void): this;
+
+        /**
+         * Must be called to finish writing the PDF document
+         */
+        end(): Promise<void>;
+
+        /**
+         * Can be used to render the document as a buffer
+         */
+        asBuffer(): Promise<Buffer>;
+
+        /**
+         * Can be used to render the document as a buffer
+         * @param callback callback called with either error or buffer
+         */
+        asBuffer(callback?: (err: Error, data: Buffer) => void): void;
+
+        /**
+         * Add a header to the document
+         */
+        header(): Header;
+
+        /**
+         * Add a footer to the document
+         */
+        footer(): Footer;
+
+        /**
+         * Add all pages of an external PDF into this document (aka merge an external document into this document)
+         * @param external The external document to merge
+         */
+        addPagesOf(external: ExternalDocument): void;
+
+        /**
+         * Add one specific page of an external PDF into this document (aka merge an external document into this document)
+         * @param page The number of the page that should be added
+         * @param external The external document to merge
+         */
+        addPageOf(page: number, external: ExternalDocument): void;
+
+        /**
+         * Use an external document as a page template (i.e. external PDF will be used as a starting point / as a background for all pages)
+         * @param external The external document to merge
+         */
+        setTemplate(external: ExternalDocument): void;
+    }
+
+    export class Fragment {
+        end(): void; // TODO
+
+        /**
+         * Add some text to the document
+         * @param text A string that should be rendered
+         * @param opts Options
+         */
+        text(text?: string, opts?: TextOptions): Text;
+
+        /**
+         * Add some text to the document
+         * @param opts Options
+         */
+        text(opts: TextOptions): Text;
```

Comment:
It is also possible to call `doc.text()` without arguments, or is this case already included in the above `text(text?: string, opts?: TextOptions): Text;`?

Reply:
It is indeed included through the `text` declaration which takes two optional parameters (example in [pdfjs-tests.ts](https://github.com/rkusa/pdfjs/pull/91/files#diff-6a30a67a6102f1341dc32166f355b90fR132)).